File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        count = 0
        i = 0
        n = len(nums)
        while i < n:
            if nums[i] != val:
                temp = nums[i]
                nums[i] = nums[count]
                nums[count] = temp
                count += 1
            i += 1

        return count

File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> curr;
        vector<bool> visited(nums.size(), false);

        backtrack(nums, curr, visited, res);
        return res;
    }

    void backtrack(vector<int>& nums, vector<int>& curr, vector<bool>& visited, vector<vector<int>>& res) {
        if (curr.size() == nums.size()) {
            res.push_back(curr);
            return;
        }

        for (int i = 0; i < nums.size(); i++) {
            if (visited[i]) continue;

            visited[i] = true;
            curr.push_back(nums[i]);

            backtrack(nums, curr, visited, res);

            curr.pop_back();
            visited[i] = false;
        }
        return;
    }
};
